### DISCUSSION TITLE
Return type of adapt() in angular

### DISCUSSION THREAD

**Discussion by bauerbua** [state-adapt/state-adapt#40] [Q&A]
Is there a return type of the adapt method from @state-adapt/angular?

```typescript
export class FileUploadStore {
    public store = adapt(['fileUploadStore', fileUploadInitialState], fileUploadAdapter)
}
```

Typescript complains about unresolved variables inside of a html file.

**Comment by mfp22** [comment]
Yeah it has a return type, but it relies on inference a lot, so defining it yourself would be annoying.

What is the exact error?

Also try telling VSCode to restart the TS server. If you just installed StateAdapt it could have issues.

**Reply by bauerbua** [reply]
I am using WebStorm unfortunately. I can work with it, so there is no compile error, but it would be nice to autocomplete the methods. Also highlighting them inside of a .ts file does not work without a defined type, but might as well be just an issue with Intellij...
```html
<ng-container *ngIf="store.filesAll$ | async as files">
...
<ng-container>
```
The typescript-error would be "Unresolved variable filesAll$"
